runhouse/main.py

```diff
@@ -590,159 +592,70 @@ def status(
-        )
-        clusters_in_den = []
-    else:
-        clusters_in_den = clusters_in_den_resp.json().get("data")
+    clusters = Cluster.list(show_all=show_all, since=since, status=cluster_status)
 
-    clusters_in_den_names = [cluster.get("name") for cluster in clusters_in_den]
+    all_clusters = clusters.get("all_clusters", None)
+    running_clusters = clusters.get("running_clusters", None)
+    sky_clusters = clusters.get("sky_clusters", None)
 
-    if not sky_clusters and not clusters_in_den:
-        console.print("No existing clusters.")
-
-    if sky_clusters:
-        # getting the on-demand clusters that are not saved in den.
-        sky_clusters = [
-            cluster
-            for cluster in sky_clusters
-            if f'/{rns_client.username}/{cluster.get("name")}'
-            not in clusters_in_den_names
-        ]
-
-    total_clusters = len(clusters_in_den)
-    table_title = f"[bold cyan]Clusters for {rns_client.username} (Total: {total_clusters})[/bold cyan]"
-
-    table = Table(title=table_title, title_justify="left")
-
-    # Add columns to the table
-    table.add_column("Name", justify="left", no_wrap=True)
-    table.add_column("Cluster Type", justify="center", no_wrap=True)
-    table.add_column("Status", justify="left")
-
-    running_clusters = []
-    not_running_clusters = []
+    if not all_clusters:
+        console.print(f"No clusters found in Den for {rns_client.username}.")
+        print_sky_clusters_msg(len(sky_clusters), console)
+        return
 
-    for den_cluster in clusters_in_den:
-        # get just name, not full rns address. reset is used so the name will be printed all in white.
-        cluster_name = f'[reset]{den_cluster.get("name").split("/")[-1]}'
-        cluster_type = den_cluster.get("data").get("resource_subtype")
-        cluster_status = (
-            den_cluster.get("status") if den_cluster.get("status") else None
-        )
+    filters_requested: bool = show_all or since or cluster_status
 
-        # currently relying on status pings to den as a sign of cluster activity.
-        # The split is required to remove milliseconds and the offset (according to UTC) from the timestamp.
-        # (status_last_checked is in the following format: YYYY-MM-DD HH:MM:SS.ssssss±HH:MM)
+    clusters_to_print = all_clusters if filters_requested else running_clusters
 
-        last_active_at = den_cluster.get("status_last_checked")
-        last_active_at = (
-            datetime.datetime.fromisoformat(last_active_at.split(".")[0])
-            if isinstance(last_active_at, str)
-            else None
-        )
-        last_active_at = (
-            last_active_at.replace(tzinfo=datetime.timezone.utc)
-            if last_active_at
-            else None
-        )
+    if show_all:
+        # if user requesting all den cluster, limit print only to 50 clusters max.
+        clusters_to_print = clusters_to_print[: (min(len(clusters_to_print), 50))]
```

Comment:
can we make 50 a constant? at the very least would allow a user to overwrite it to something larger if they want in the meantime